### PR TITLE
Beta TC VAE

### PR DESCRIPTION
**Pull request by jaceybronte**
This PR implements the Beta TC VAE and cleans up the train-VAE folder

### CODE REVIEW

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: Consider making these in alphabetical order using PEP8 import standards. 

Might consider using isort with pre-commit : https://github.com/WayScience/MNIST_TD-VAE/blob/master/.pre-commit-config.yaml

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py:63]: Nice docstring! :D

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: Consider making this an optional arg with these dimension as the default. This will cut down on lines of code. I will make the suggestion above

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]:
```suggestion
                 hidden_dims: List =  [256, 256, 256],
```

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: Consider not hardcoding the 256 here and using a variable instead

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: Consider logging instead of print statements here

Comment (MikeLippincott) [reply 2.train-VAE/betatcvae.py]: Might be an issue for another PR though?

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py:118]: consider adding docstrings to these methods

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: generally, all caps variables are global vars. Consider refactoring this one.

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py:209]: consider adding doc strings and type hints here

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: consider making this path an arg for the function

Comment (MikeLippincott) [reply 2.train-VAE/betatcvae.py]: We want these functions to be flexible and reusable across analysis, and repos even

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py]: same here

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py:209]: Also consider splitting these up into different `.py` files. I could see a `tcvae_model.py` file for the class and a `tcvae_train_utils.py` file for the training functions.

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py:346]: Is there a way to make this programmatic to extract weights from other layers too?

File: 2.train-VAE/betatcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/betatcvae.py:151]: This is a complex loss function. I think some inline documentation might be useful for others to understand better. Consider adding comments explaining what each term is and what it is doing.

File: 2.train-VAE/optimize_utils_tcvae.py
Comment (MikeLippincott) [inline 2.train-VAE/optimize_utils_tcvae.py:180]: No SGD?

File: 2.train-VAE/scripts/3.beta_tc_vae_training_extract_model_components .py
Comment (MikeLippincott) [inline 2.train-VAE/scripts/3.beta_tc_vae_training_extract_model_components .py:23]: same here, consider using isort

File: 2.train-VAE/scripts/3.beta_tc_vae_training_extract_model_components .py
Comment (MikeLippincott) [inline 2.train-VAE/scripts/3.beta_tc_vae_training_extract_model_components .py:60]: nice!

**Review by MikeLippincott — APPROVED** [review]
Good job, ready to merge after addressing comments. Let me know if you want help setting up pre-commit! LGTM.

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: Fixed

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py:63]: Thanks!

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: fixed

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: fixed

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: fixed

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: this was just to find a matrix multiplication error, removed!

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py:118]: done

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: fixed

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py:209]: fixed

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: done

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py]: done

File: 2.train-VAE/scripts/3.beta_tc_vae_training_extract_model_components .py
Comment (jaceybronte) [reply 2.train-VAE/scripts/3.beta_tc_vae_training_extract_model_components .py:23]: fixed

File: 2.train-VAE/optimize_utils_tcvae.py
Comment (jaceybronte) [reply 2.train-VAE/optimize_utils_tcvae.py:180]: just not giving any sort of good results and adjusting other hyperparams to fix has negative effects on adam and rmsprop

File: 2.train-VAE/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/betatcvae.py:151]: yep

**Pull request merged**
